### PR TITLE
Object selection

### PR DESCRIPTION
**Pull request by cbrightsci**
Just for you @dcwhite :)

### CODE REVIEW

**Comment by dcwhite** [comment]
Excited to test this!

**Comment by aylakhan** [comment]
Looks good in tikhonov-inverse network!

File: src/Interface/Modules/Render/ViewScene.h [hunk 1]
```diff
@@ -90,6 +92,7 @@ namespace SCIRun {
       virtual void hideEvent(QHideEvent* evt) override;
       virtual void contextMenuEvent(QContextMenuEvent* evt) override {}
     private:
+      bool isObjectUnselected(std::string name);
```
Comment (dcwhite) [inline src/Interface/Modules/Render/ViewScene.h]: Change to const std::string&

File: src/Interface/Modules/Render/ViewScene.cc [hunk 2]
```diff
@@ -47,7 +47,7 @@ using namespace SCIRun::Render;
 ViewSceneDialog::ViewSceneDialog(const std::string& name, ModuleStateHandle state,
 	QWidget* parent /* = 0 */)
 	: ModuleDialogGeneric(state, parent), mConfigurationDock(0), shown_(false),
-	itemManager_(new ViewSceneItemManager)
+	itemManager_(new ViewSceneItemManager(state, this))
```
Comment (dcwhite) [inline src/Interface/Modules/Render/ViewScene.cc]: Passing this in initialization list of constructor is risky. I'd pull that out and use a setter inside the constructor body

File: src/Interface/Modules/Render/ViewScene.cc [hunk 3]
```diff
@@ -390,6 +399,33 @@ void ViewSceneDialog::assignBackgroundColor()
...
+void ViewSceneDialog::handleSelectedItem(const QString& name)
+{
+  unselectedObjectNames_.erase(std::remove(unselectedObjectNames_.begin(), unselectedObjectNames_.end(), name.toStdString()), unselectedObjectNames_.end());
+  newGeometryValue();
+}
+
+//------------------------------------------------------------------------------
+bool ViewSceneDialog::isObjectUnselected(std::string name)
+{
+  for (auto it = unselectedObjectNames_.begin(); it != unselectedObjectNames_.end(); ++it)
```
Comment (dcwhite) [inline src/Interface/Modules/Render/ViewScene.cc]: This manual loop can be replaced with an std::find call
```
return std::find(unselectedObjectNames_.begin(); unselectedObjectNames_.end(), name) != unselectedObjectNames_.end();
```

File: src/Interface/Modules/Render/ViewScene.cc [hunk 4]
```diff
@@ -390,6 +399,33 @@ void ViewSceneDialog::assignBackgroundColor()
 }
 
 //------------------------------------------------------------------------------
+void ViewSceneDialog::handleUnselectedItem(const QString& name)
+{
+  unselectedObjectNames_.push_back(name.toStdString());
+  newGeometryValue();
```
Comment (dcwhite) [inline src/Interface/Modules/Render/ViewScene.cc]: This may be slow when user clicks a checkbox many times in a row. I'll test that out.

**Comment by dcwhite** [comment]
Works OK in networks with a single ViewScene. Having problem with multiple ViewScenes, report soon.

**Comment by dcwhite** [comment]
False alarm transparency was set to 0 :P

File: src/Interface/Modules/Render/ViewScene.cc (hunk 4 above)
Comment (dcwhite) [inline src/Interface/Modules/Render/ViewScene.cc]: Heh, the combo box we're using doesn't allow a ton of clicks in a row, preventing this goofy behavior. We can examine the performance later

**Pull request merged**